api/v1alpha1/datadogagent_default.go

```diff
@@ -902,7 +909,10 @@ func DefaultDatadogFeaturePrometheusScrape(ft *DatadogFeatures) *PrometheusScrap
 func DefaultDatadogFeatureNetworkMonitoring(ft *DatadogFeatures) *NetworkMonitoringConfig {
 	if ft.NetworkMonitoring == nil {
 		ft.NetworkMonitoring = &NetworkMonitoringConfig{Enabled: NewBoolPointer(false)}
-		return ft.NetworkMonitoring
+
+		if !BoolValue(ft.NetworkMonitoring.Enabled) {
```

Comment:
This seems like it'll always be the case, why adding it?

Reply:
if never we change the default `Enabled` value. 
I tried to use the same pattern on every `Default...` fonction